### PR TITLE
better cancelled error handling

### PR DESCRIPTION
**Pull request by vangheem**

### CODE REVIEW

File: guillotina/traversal.py [hunk 1]
```diff
@@ -399,9 +400,16 @@ def set_root(self, root: Optional[IApplication]):
         try:
             result = await self.real_resolve(request)
         except (response.Response, aiohttp.web_exceptions.HTTPException) as exc:
+            _clean_request(request, exc)
```
Comment (masipcat) [inline guillotina/traversal.py]: This is already doing here: https://github.com/plone/guillotina/pull/785/files#diff-4decc550a3956a0cf87e7f196cac7d40L365

File: guillotina/traversal.py [hunk 2]
```diff
@@ -399,9 +400,16 @@ def set_root(self, root: Optional[IApplication]):
...
             await abort()
             return BasicMatchInfo(request, exc)
+        except asyncio.CancelledError:
+            _clean_request(request, None)
```
Comment (masipcat) [inline guillotina/traversal.py]: same

**Comment by vangheem** [comment]
@masipcat thanks, fixed.

File: guillotina/tests/test_errors.py [hunk 3]
```diff
@@ -29,3 +32,25 @@ def test_value_serialization_error():
...
+
+async def test_handle_cancelled_error(container_requester):
+    async with container_requester as requester:
+        with mock.patch("guillotina.api.content.DefaultGET.__call__") as handle_mock:  # noqa
+            f = asyncio.Future()
+            f.set_result(None)
+            handle_mock.return_value = f
+            handle_mock.side_effect = asyncio.CancelledError()
+            response, status = await requester("GET", "/db/guillotina")
+            status == 499
```
Comment (masipcat) [inline guillotina/tests/test_errors.py]: You forgot to assert :P I didn't understand why the coverage decreased...

Comment (vangheem) [reply guillotina/tests/test_errors.py]: eeek. I think we have multiple tests with this. I'll check again.

Comment (vangheem) [reply guillotina/tests/test_errors.py]: fixed here: https://github.com/plone/guillotina/pull/786

**Pull request merged**